### DISCUSSION TITLE
Testing midi availablity in jest

### DISCUSSION THREAD

**Discussion by glend1** [djipco/webmidi#240] [Migration]
Since the library now supports midi in node my testings is failing because i don't know how to test if midi is unavailable. 

any ideas how i can stop midi working for testing?

**Comment by djipco** [comment]
The [tests](https://github.com/djipco/webmidi/tree/master/test) for the library itself rely upon the `midi` Node.js module. This module allows you to create and destroy MIDI ports at will and should work for your needs. You can check out the [WebMidi.test.js](https://github.com/djipco/webmidi/blob/master/test/WebMidi.test.js) file for an example.

**Comment by glend1** [comment]
Have you tried using [midi-test from jazz-soft](https://github.com/jazz-soft/midi-test)? 

It has fewer requirements than [node-midi](https://www.npmjs.com/package/midi) because it mocks the midi implementation. It might be able to fix some of those commented out tests you have in that file. Should (theoretically) be faster too.

I've been trying to use it and I have hit some what of a brickwall (due to the way it possibly handles promises) so im curious if you've experimented with it and ditched it.

**Comment by djipco** [comment]
I have not tried it. I believe it had not yet been released when I chose to go with the `midi` package. I'm open to using it but, unfortunately, I have no time to dedicate to this right now. Let me know if you are ever able to fix the issues you are facing with it.
